include/boost/geometry/formulas/vertex_latitude.hpp

```diff
@@ -0,0 +1,149 @@
+// Boost.Geometry
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_MAXIMUM_LATITUDE_HPP
+#define BOOST_GEOMETRY_FORMULAS_MAXIMUM_LATITUDE_HPP
+
+#include <boost/geometry/formulas/spherical.hpp>
+
+namespace boost { namespace geometry { namespace formula
+{
+
+/*!
+\brief Algorithm to compute the vertex latitude of a geodesic segment. Vertex is
+a point on the geodesic that maximizes (or minimizes) the latitude.
+\author See
+    Wood - Vertex Latitudes on Ellipsoid Geodesics, SIAM Rev., 38(4), 637–644,
+    1996
+*/
+
+template <typename CT>
+class vertex_latitude
+{
+
+public:
+
+    struct vertex_lat_result
+    {
+        vertex_lat_result()
+            :   north(0),
+              south(0)
+        {}
+
+        CT north, south;
+    };
+
+    template <
+            typename T1,
+            typename T2
+            >
+    static inline vertex_lat_result spherical(T1 const& lon1,
+                                              T1 const& lat1,
+                                              T2 const& lon2,
+                                              T2 const& lat2)
+    {
+        vertex_lat_result vrt_result;
+
+        geometry::formula::result_spherical<CT> result = geometry::formula::
+                spherical_azimuth<CT, true>(lon1, lat1, lon2, lat2);
+
+        CT const alp1 = std::abs(result.azimuth);
+        CT const alp2 = std::abs(result.reverse_azimuth);
+
+        if(vertex_on_segment(alp1, alp2, lat1, lat2, vrt_result))
```

Comment:
minor: I think we have spaces after if, so if ( 
(contrary to function calls where we don't do that)